src/_pytest/nodes.py

```diff
@@ -340,6 +341,8 @@ def _repr_failure_py(
             return excinfo.value.formatrepr()
         if self.config.getoption("fulltrace", False):
             style = "long"
+        if excinfo.type is ConftestImportFailure:  # type: ignore
+            excinfo = ExceptionInfo.from_exc_info(excinfo.value.excinfo)  # type: ignore
```

Comment:
I believe the `excinfo` parameter should be changed to also include ConftestImportFailure in the Union and that'll make mypy happy